applications/rag/README.md

```diff
@@ -1,192 +1,204 @@
 
-By default, Terraform uses the pre-existing VPC network (`default`) in your GCP project.
-CloudSQL requires [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect) to be enabled in the VPC network.
+If you prefer to use your own VPC, set `create_network = false` in the following section under `workloads.tfvars`. This also requires enabling [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect) for your VPC. Without Private Service Connect, the RAG components cannot connect to the vector DB:
 
 1. [Create an IP allocation](https://cloud.google.com/vpc/docs/configure-private-services-access#procedure)
 
 2. [Create a private connection](https://cloud.google.com/vpc/docs/configure-private-services-access#creating-connection).
 
-### Deploy terraform components
+## Installation
+
+This section sets up the RAG infrastructure in your GCP project using Terraform.
 
-The following steps set up the cluster, inference server, pgvector CloudSQL instance, JupyterHub, KubeRay, and frontend chat interface.
+**NOTE:** Terraform keeps state metadata in a local file called `terraform.tfstate`. Deleting the file may cause some resources to not be cleaned up correctly even if you delete the cluster. We suggest using `terraform destroy` before reapplying/reinstalling.
 
 1. `cd ai-on-gke/applications/rag`
 
-2. Edit `workloads.tfvars` to set your project ID, location, cluster name, and GCS bucket name. Optionally, make the following changes:
-    * Set `create_cluster = false` if you are using an existing cluster.
-     * (Recommended) Set `jupyter_add_auth = true`, `frontend_add_auth = true` and `ray_dashboard_add_auth = true` to create load balancers with IAP for your Jupyter notebook, TGI frontend and Ray Dashboard.
-    * (Recommended) Set `jupyter_domain`, `frontend_domain`, `ray_dashboard_domain` to specify a domain, you need to be able to configure DNS records of that domain, you can [register a Domain on Google Cloud Domains](https://cloud.google.com/domains/docs/register-domain#registering-a-domain), or use the domain registrar of your choice.
-    * Choose a custom k8s namespace and service account to be used by the application.
-    * Set `create_network = true` if you want to create a new VPC network
+2. Edit `workloads.tfvars` to set your project ID, location, cluster name, and GCS bucket name. Ensure the `gcs_bucket` name is unique across all GCP projects (add a random suffix). Optionally, make the following changes:
+    * (Optional) Set a custom `kubernetes_namespace` where all k8s resources will be created.
+    * (Recommended) Enable authenticated access for JupyterHub, frontend chat and Ray dashboard services:
+        - First, make sure the [OAuth Consent Screen](https://developers.google.com/workspace/guides/configure-oauth-consent#configure_oauth_consent) is configured for your project. Ensure `User type` is set to `Internal`.
```

Comment:
Can we move the IAP/custom domain instructions to an appendix section? These steps are quite long and clutter the instructions.